pkg/tfbridge/auto_aliasing.go

```diff
@@ -369,11 +377,17 @@ func applyMaxItemsOneAliasing(schema shim.Schema, h *fieldHistory, info *SchemaI
 	switch e := e.(type) {
 	case shim.Resource:
 		populateElem()
-		eHasH, eHasI := applyResourceMaxItemsOneAliasing(e, &h.Elem.Fields, &info.Elem.Fields)
+		var eHasH, eHasI bool
+		if h.Elem != nil {
+			eHasH, eHasI = applyResourceMaxItemsOneAliasing(e, &h.Elem.Fields, &info.Elem.Fields)
```

Comment:
Are we sure this here is safe? These functions edit SchemaInfo.MaxItemsOne to ensure backwards compatibility. If these edits are dropped at runtime, but only performed at compile-time, we'll get unexpected runtime serializer errors possibly.

Reply:
`applyResourceMaxItemsOneAliasing` and `applyMaxItemsOneAliasing` panic if  `h.Elem` is `nil`

The change in line 368 above means that sometimes `h.elem` will be `nil` here (ie. in the case that there was nothing in the metadata for this field), so we have to add the test. At runtime, it's safe to just skip because there's nothing in the metadata.

Although, it writing this I can see a cleaner way to handle this with a lot fewer conditionals, so I'll try that.

Reply:
I'm worried that this changes semantics. I don't know what conditions precisely get us to "sometimes h.elem will be nil" but it sounds like it shouldn't be? 

Are we quite certain that under both versions of the flag `isTfgen() = {true,false}` we're getting the *same* providerInfo back?

Something perhaps checkable with:

```
reflect.DeepEqual(a.Resources, b.Resouces)`
```

If this holds on a big provider I'd be less worried.

Reply:
Hm the last version looks a lot cleaner! It seems to me only history edits are skipped, and there's short-cicuiting of edits when we know there's no history so there definitionally shouldn't be edits. This looks safer. Think about the invariant above but if you're confident we don't need further testing.

Reply:
I'll run that test on aws before I merge, just to be sure.

Reply:
So the Resources are not quite identical, but I think semantically the same. When running with `isTfGen() == false` we sometimes produce empty field maps, where the version with `isTfgen() == true` produces `nil`
```
        	            	Diff:
        	            	--- Expected
        	            	+++ Actual
        	            	@@ -79007,4 +79007,3 @@
        	            	      Elem: (*tfbridge.SchemaInfo)(<nil>),
        	            	-     Fields: (map[string]*tfbridge.SchemaInfo) {
        	            	-     },
        	            	+     Fields: (map[string]*tfbridge.SchemaInfo) <nil>,
        	            	      Asset: (*tfbridge.AssetTranslation)(<nil>),
        	            	@@ -79124,4 +79123,3 @@
        	            	      Elem: (*tfbridge.SchemaInfo)(<nil>),
        	            	-     Fields: (map[string]*tfbridge.SchemaInfo) {
        	            	-     },
        	            	+     Fields: (map[string]*tfbridge.SchemaInfo) <nil>,
        	            	      Asset: (*tfbridge.AssetTranslation)(<nil>),
        	            	@@ -79168,4 +79166,3 @@
        	            	      Elem: (*tfbridge.SchemaInfo)(<nil>),
        	            	-     Fields: (map[string]*tfbridge.SchemaInfo) {
        	            	-     },
        	            	+     Fields: (map[string]*tfbridge.SchemaInfo) <nil>,
        	            	      Asset: (*tfbridge.AssetTranslation)(<nil>),
        	            	@@ -79212,4 +79209,3 @@
        	            	      Elem: (*tfbridge.SchemaInfo)(<nil>),
        	            	-     Fields: (map[string]*tfbridge.SchemaInfo) {
        	            	-     },
        	            	+     Fields: (map[string]*tfbridge.SchemaInfo) <nil>,
        	            	      Asset: (*tfbridge.AssetTranslation)(<nil>),
        	Test:       	TestProviderSame
```

Reply:
Yes, nil vs empty should be fine here.